webfront/views/cache.py

```diff
@@ -6,11 +6,66 @@
 
 from rest_framework import status
 from webfront.response import Response
+from webfront.views.utils import endpoints
+
 from django.conf import settings
 
 multiple_slashes = re.compile("/+")
 
 
+FIVE_DAYS = 5 * 24 * 60 * 60
+SHOULD_NO_CACHE = -1
+
+names = [ep["name"].lower() for ep in endpoints]
+
+
+def get_timeout_from_path(path, endpoint_levels):
+    parsed = urlparse(path)
+    # process query
+    query = parse_qs(parsed.query, keep_blank_values=True)
+
+    if (  # is requesting by accession
+        len(endpoint_levels) == 3
+        and len([ep for ep in endpoint_levels if ep in names]) < 2
```

Comment:
Shouldn't `ep` be converted to lower case as well?

Also, I am not sure I understand why we check whether the length is `< 2`. Why not (`== 1`)? In which case(s) `endpoint_levels` can contain no top-level endpoint block?

Reply:
> Shouldn't `ep` be converted to lower case as well?

Cool, I'll add the conversion.

> Also, I am not sure I understand why we check whether the length is `< 2`. Why not (`== 1`)? In which case(s) `endpoint_levels` can contain no top-level endpoint block?

TBH I have no idea why I put `<2` it really seems like a conscious decision to put it that way. But from what I can see it really seems that `==1` is better here. 🤷🏽